include/drivers/spi.h

```diff
@@ -162,6 +162,11 @@ struct spi_cs_control {
  *
  * @note Only cs_hold and lock_on can be changed between consecutive
  * transceive call. Rest of the attributes are not meant to be tweaked.
+ *
+ * @warning Most drivers use pointer comparison to determine whether a
```

Comment:
Would it help the user here if we mention `spi_context_configured()` explicitly? The pointer comparison is a bit hidden, so maybe some users might not notice if they look in the driver's source to check for this warning. But on the other hand, we should maybe not mention internal details in the API documentation since it can go out-of-date.